extensions/amp-script/0.1/test-e2e/test-amp-script.js

```diff
@@ -73,5 +71,27 @@ describes.endtoend(
 
       await expect(realRatio).to.equal(targetRatio);
     });
+
+    // In layout=container, amp-script requires mutations to be backed by
+    // user gestures. This ensures that this requirement is also enforced
+    // on load AKA "hydration".
+    it('should not mutate on load in layout=container', async () => {
+      const element = await controller.findElement('amp-script#mutate');
+      await expect(controller.getElementAttribute(element, 'class')).to.contain(
+        'i-amphtml-hydrated'
+      );
+
+      // `document.body.textContent = lipsum;` should be disallowed.
+      await expect(controller.getElementText(element)).to.contain(
+        'Append some very long text'
+      );
+
+      // However, gesture-backed (e.g. click) mutations are OK.
+      const button = await controller.findElement('#mutate button');
+      controller.click(button);
+
+      const h1 = await controller.findElement('#mutate h1');
+      await expect(controller.getElementText(h1)).to.contain('Lorem Ipsum');
```

Comment:
only the first part of this test is actually for the 'hydration' phase. The 'click' occurs afterwards, and is really testing that the event listener was added.  I think thats fine, since the second half is still useful (test that adding event listeners still works)

Reply:
Yea, as you mention the distinction for "hydration" is not really useful. It's only mentioned here due to the bug.